### DISCUSSION TITLE
[FR] Clearing Unused assets

### DISCUSSION THREAD

**Discussion by michtio** [craftcms/cms#8736] [Ideas]
### Description

We are currently running an asset heavy multisite setup ( ~36 sites setup in the sites section ) so those different sites do share assets through one another, but do have different entries and not the same content written at all times.

There are a few thousands of assets on the production site currently, and it would be easy to have a CLI command that checks through all assets and deletes the ones which aren't in use in any entry/site anymore. This to keep the assets clean and also reduce database load and more. 

Why a FR? I see that this can be a very useful addition to Craft CMS to keep things optimized and could be used as part of a Deployment script or Cron Job through the CLI.

**Comment by arifje** [comment]
We could use this too. In the last few years assets that are replaced / removed in an entry are piling up, taking a lot of space.

I created a small plugin with a console command a white ago, just added it to this public repo: https://github.com/arifje/craft-asset-cleanup - maybe you can use it too, or create your own plugin from it (we're using Transcoder plugin too, and this also cleans up some generated thumbnails, but it's hard coded).

ps. it also contains a test mode, so nothing will be deleted, but only scanned.
